tools/src/main/java/org/apache/kafka/tools/TopicCommand.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.MutuallyExclusiveGroup;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.ListTopicsOptions;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicListing;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.apache.kafka.common.internals.Topic;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import static net.sourceforge.argparse4j.impl.Arguments.storeTrue;
+
+/**
+ * Command class for operations on topics
+ */
+public class TopicCommand {
+
+    private AdminClient client;
+
+    public static void main(String[] args) throws Exception {
+
+        new TopicCommand().run(args);
+    }
+
+    void run(String[] args) throws Exception {
+
+        TopicCommandOptions opts = new TopicCommandOptions(args);
+
+        if (args.length == 0)
+            CommandLineUtils.printUsageAndDie(opts.parser, "Create, delete, describe, or change a topic.");
+
+        opts.checkArgs();
+
+        Properties props = new Properties();
+        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, opts.brokerList());
+
+        this.client = AdminClient.create(props);
+
+        try {
+
+            if (opts.has("create")) {
+                this.createTopics(opts);
+            } else if (opts.has("alter")) {
+                // TODO
+            } else if (opts.has("list")) {
+                this.listTopics(opts);
+            } else if (opts.has("describe")) {
+                // TODO
+            } else if (opts.has("delete")) {
+                // TODO
+            }
+
+        } catch (ArgumentParserException e) {
```

Comment:
In this case, in my understanding, the `run` method should be in the base abstract class with a specific structure and then having derived class implementing some abstract methods. I.e. in the abstract class having : 

`void run( ...) {`
`doCheckArgs()`
` // admin client creation and configuration`
`try {`
`doCommand()`
`} catch (...) {`
`....`
`} finally {`
`....`
`}`

where `doCheckArgs` and `doCommand` are abstract methods that the derived class has to implement ... but are we sure that this fixed structure can work for all the tools ?